src/containerapp/azext_containerapp/tests/latest/test_containerapp_patch.py

```diff
@@ -0,0 +1,204 @@
+        image_name = app["properties"]["template"]["containers"][0]["image"]
+        current_image = client.images.get(image_name)
+        registry = app["properties"]["configuration"]["registries"][0]["server"]
+        repo = app["properties"]["template"]["containers"][0]["name"]
+
+        # Re-tag image to an older version to simulate patch list
+        old_tag = "run-dotnet-aspnet-7.0.1-cbl-mariner2.0"
+        old_image_name = registry + "/" + repo + ":" + old_tag
+        current_image.tag(registry + "/" + repo, tag = old_tag)
+
+         # Re-tag and push
+        client.images.push(registry + "/" + repo, tag = old_tag)
+        self.cmd('az containerapp update -n {} -g {} --image {}'.format(app_name, resource_group, old_image_name))
+        app = self.cmd(f"containerapp show -g {resource_group} -n {app_name}").get_output_in_json()
+        retagged_image = app["properties"]["template"]["containers"][0]["image"]
+        retagged_image_tag = retagged_image.split(':')[1]
+
+        # Verify if image was updated
+        self.assertEquals(retagged_image_tag,old_tag)
+
+    def _create_and_verify_containerapp_up(
+            self,
+            resource_group,
+            env_name = None,
+            source_path = None,
+            image = None,
+            location = None,
+            ingress = None,
+            target_port = None,
+            app_name = None):
+        # Configure the default location
+        self.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+
+        # Ensure that the Container App environment is created
+        if env_name is None:
+           env_name = self.create_random_name(prefix='env', length=24)
+           self.cmd(f'containerapp env create -g {resource_group} -n {env_name}')
+
+        if app_name is None:
+            # Generate a name for the Container App
+            app_name = self.create_random_name(prefix='containerapp', length=24)
+
+        # Construct the 'az containerapp up' command
+        up_cmd = f"containerapp up -g {resource_group} -n {app_name} --environment {env_name}"
+        if source_path:
+            up_cmd += f" --source \"{source_path}\""
+        if image:
+            up_cmd += f" --image {image}"
+        if ingress:
+            up_cmd += f" --ingress {ingress}"
+        if target_port:
+            up_cmd += f" --target-port {target_port}"
+
+        # Execute the 'az containerapp up' command
+        self.cmd(up_cmd)
+
+        # Verify that the Container App is running
+        app = self.cmd(f"containerapp show -g {resource_group} -n {app_name}").get_output_in_json()
+        url = app["properties"]["configuration"]["ingress"]["fqdn"]
+        url = url if url.startswith("http") else f"http://{url}"
+        resp = requests.get(url)
+        self.assertTrue(resp.ok)
+
+        # Re-run the 'az containerapp up' command with the location parameter if provided
+        if location:
```

Comment:
I know this part is similar to the [up command test case](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/tests/latest/test_containerapp_up.py#LL92C1-L95C29)., but could you clarify the reason why we need to rerun the command with location argument instead of directly providing this as an argument?